### PR TITLE
added unit tests for postgres connector

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+	b := make([]byte, n)
+	_, err := rand.Read(b)
+	suite.failTestError(err)
+	return b
+}
+
+func (suite *PostgresCDCTestSuite) randString(n int) string {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+	rand.Int()
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:96]: spurious warning for the usecase, ignoring wherever applicable

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+		suite.connector.GetLastOffset("offset_panic")
+	}, "not implemented")
+	suite.Panicsf(func() {
+		suite.connector.GetLastSyncBatchID("sync_batch_id_panic")
+	}, "not implemented")
+	suite.Panicsf(func() {
+		suite.connector.GetLastNormalizeBatchID("normalize_batch_id_panic")
+	}, "not implemented")
+	suite.Panicsf(func() {
+		suite.connector.GetDistinctTableNamesInBatch("distinct_table_names_in_batch_panic", 0, 0)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:286]: we want to test for a panic here, ignoring wherever when applicable

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+	_, err = suite.connector.pool.Exec(context.Background(),
+		fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s(id INT PRIMARY KEY, name TEXT)", nonExistentFlowSrcTableName))
+	suite.failTestError(err)
+	ensurePullabilityOutput, err = suite.connector.EnsurePullability(&protos.EnsurePullabilityInput{
+		FlowJobName:           nonExistentFlowName,
+		SourceTableIdentifier: nonExistentFlowSrcTableName,
+		PeerConnectionConfig:  nil, // not used by the connector itself.
+	})
+	suite.failTestError(err)
+	tableRelId := ensurePullabilityOutput.TableIdentifier.GetPostgresTableIdentifier().RelId
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:404]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+			allTypesHappyFlowSrcTableName))
+	suite.failTestError(err)
+
+	ensurePullabilityOutput, err := suite.connector.EnsurePullability(&protos.EnsurePullabilityInput{
+		FlowJobName:           allTypesHappyFlowName,
+		SourceTableIdentifier: allTypesHappyFlowSrcTableName,
+		PeerConnectionConfig:  nil, // not used by the connector itself.
+	})
+	suite.failTestError(err)
+	tableRelId := ensurePullabilityOutput.TableIdentifier.GetPostgresTableIdentifier().RelId
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:555]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+		fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s(id INT PRIMARY KEY, name TEXT)", simpleHappyFlowSrcTableName))
+	suite.failTestError(err)
+
+	ensurePullabilityOutput, err := suite.connector.EnsurePullability(&protos.EnsurePullabilityInput{
+		FlowJobName:           simpleHappyFlowName,
+		SourceTableIdentifier: simpleHappyFlowSrcTableName,
+		PeerConnectionConfig:  nil, // not used by the connector itself.
+	})
+	suite.failTestError(err)
+	tableRelId := ensurePullabilityOutput.TableIdentifier.GetPostgresTableIdentifier().RelId
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:446]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+	_, err = setupTx.Exec(context.Background(), "CREATE SCHEMA pgpeer_test")
+	suite.failTestError(err)
+	err = setupTx.Commit(context.Background())
+	suite.failTestError(err)
+}
+
+func (suite *PostgresCDCTestSuite) TearDownSuite() {
+	teardownTx, err := suite.connector.pool.Begin(context.Background())
+	suite.failTestError(err)
+	defer teardownTx.Rollback(context.Background())
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:238]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+		Port:     7132,
+		User:     "postgres",
+		Password: "postgres",
+		Database: "postgres",
+	})
+	suite.failTestError(err)
+
+	setupTx, err := suite.connector.pool.Begin(context.Background())
+	suite.failTestError(err)
+	defer setupTx.Rollback(context.Background())
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:226]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 8]
```diff
@@ -0,0 +1,739 @@
...
+		suite.Equal(32768, len(insertRecord.Items["lz4_t"].(string)))
+		suite.Equal(32768, len(insertRecord.Items["n_b"].([]byte)))
+		suite.Equal(32768, len(insertRecord.Items["lz4_b"].([]byte)))
+	}
+}
+
+func (suite *PostgresCDCTestSuite) mutateToastRecords(srcTableName string) {
+	mutateRecordsTx, err := suite.connector.pool.Begin(context.Background())
+	suite.failTestError(err)
+	defer mutateRecordsTx.Rollback(context.Background())
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:140]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 9]
```diff
@@ -0,0 +1,739 @@
...
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+	return string(b)
+}
+func (suite *PostgresCDCTestSuite) insertToastRecords(srcTableName string) {
+	insertRecordsTx, err := suite.connector.pool.Begin(context.Background())
+	suite.failTestError(err)
+	defer insertRecordsTx.Rollback(context.Background())
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:106]: changed

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 10]
```diff
@@ -0,0 +1,739 @@
...
+		MaxBatchSize:           100,
+		SrcTableIDNameMapping:  relIDTableNameMapping,
+		TableNameMapping:       tableNameMapping,
+		TableNameSchemaMapping: tableNameSchemaMapping,
+	})
+	suite.Nil(records)
+	suite.Errorf(err, "publication peerflow_pub_%s does not exist", nonExistentFlowName)
+
+	err = suite.connector.PullFlowCleanup(nonExistentFlowName)
+	suite.Errorf(err, "error dropping replication slot: ERROR: replication slot \"%s\" does not exist (SQLSTATE 42704)", nonExistentFlowName)
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:392]: changed

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**